fastcrawler/test_utils.py

```diff
@@ -0,0 +1,62 @@
+from fastcrawler.engine import AioHttpEngine
+
+
+class TestServer:
+    def __init__(self):
+        self.responses = {
+            "GET": {"message": "GET request received"},
+            "POST": {"message": "POST request received"},
+            "PUT": {"message": "PUT request received"},
+            "DELETE": {"message": "DELETE request received"},
+        }
+
+    def get_response(self, method, data=None):
+        response = self.responses.get(method)
+        if data:
+            response["data"] = data
+        return response
+
+
+class TestClient(AioHttpEngine):
```

Comment:
Never subclass from actual impl in mocking. always use Protocols or ABC instead!